src/game/moveFinding/findBishopBaseMoves.js

```diff
@@ -1,3 +1,96 @@
-export default function findBishopBaseMoves() {
-    return [];
+
+import BoardCoordinates from "../board/boardCoordinates";
+
+export default function findBishopBaseMoves(boardSituation, startingCoordinates, color) {
+    let currentBoard = boardSituation.position;
+    return checkMoves(currentBoard, startingCoordinates, color);
+
+}
+
+function checkMoves(currentBoard, startingCoordinates, color) {
+
+    let posibleMoves = []; //Array with all possible moves
+
+    //Change code to similar like rook piece 
+
+    //4 loop while for check every diagonal 
+    // on the next revision try to do with one function
+    let x = startingCoordinates.x;
+    let y = startingCoordinates.y;
+    x++;
+    y++;
+    while (x > 0 && x <= 8 && y > 0 && y <= 8) {
+        if (currentBoard[x][y] == null) {
+            posibleMoves.push(new BoardCoordinates(x, y));
+        }
+
+        else {
+            //check if you can capture this piece, if you can it is a legal move
+            if (currentBoard[x][y].color != color) {
+                posibleMoves.push(new BoardCoordinates(x, y));
+            }
+            break;
```

Comment:
To nie jest coś koniecznie wymagające poprawy ale - ogólnie rzecz biorąc - dobrym zwyczajem jest unikanie używania instrukcji "break" do standardowego wychodzenia z pętli. Kiedy deklarujesz pętlę, sama jej konstrukcja zawiera informację o tym, kiedy pętla się powinna kończyć. Zaszyte w ciele pętli instrukcje break oznaczają "ukryte" warunki wyjścia - bez wczytania się w ciało pętli, osoba czytająca Twój kod może nawet nie domyślić się, że pętla może zakończyć się "przedwcześnie" (z powodów niezwiązanych z deklarowanym warunkiem powtórzenia).

Nie musisz tego koniecznie poprawiać, ale dobrze mieć na uwadze w przyszłości. Czasem break jest przydatny i pozwala prosto napisać rzeczy, które w innych warunkach byłyby bardziej kłopotliwe (i jeszcze mniej czytelne), ale zawsze warto się zastanowić, czy naprawdę jest potrzebny.

Reply:
Dziękuje za uwagę, nie spotkałem się jeszcze z taką uwagą odnośnie break, ale masz rację na przyszłość zapamiętam :) ( Człowiek się uczy całe życie)